sdk/storage/Azure.Storage.Common.DataMovement/src/FilesystemScanner.cs

```diff
@@ -5,108 +5,142 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Text;
+using System.Linq;
 
 namespace Azure.Storage.Common.DataMovement
```

Comment:
How did we come up with this namespace and package name?
Are we going to have DataMovement per storage service + common?
If so then should this class be internal shared source?

Reply:
DMLib is more meant to be an extension package, that's how I came up with the namespace. But I'm open to any suggestions about this.

Yes the plan is to have it per storage service (exception of Queues and Blob Batch).

I don't think this should be an internal shared source. Maybe most of the classes will be internal but some of it like classes like TransferManager and the progress handler / pause and resume will be public. I also wanted to cover the scenario if a customer wants to add File Share Jobs along with Blob Jobs and/or datalake jobs.

Reply:
I definitely think we should engage the Arch Board on this and see what they think.

Reply:
@amnguye in what scenario would we expose FilesystemScanner publicly ?

Reply:
.. I definitely misread your last question woops (I was thinking the namespace for some wacky reason).

You're right FileSystemScanner should be internal.

Reply:
Great. I guess when we get to the point where we work on publicly available surface the commons shape will finalize. We're good for now.